cmd/node/main.go

```diff
@@ -1836,6 +1846,11 @@ func processDestinationShardAsObserver(prefsConfig config.PreferencesConfig) (ui
 	if len(destShard) == 0 {
 		return 0, errors.New("option DestinationShardAsObserver is not set in prefs.toml")
 	}
+
+	if destShard == notSetDestinationShardID {
+		return core.DisabledShardIDAsObserver, nil
```

Comment:
instead of this why not return 0 directly here ? after returning from this function you always change to 0 if the result was core.DisabledShardIDAsObserver